### ISSUE TITLE
webkitgtk 2.44.3 breaks UI

### ISSUE THREAD

**Issue by Psykar** [gitbutlerapp/gitbutler#4685]
### Version

0.12.19

### Operating System

Linux

### Distribution Method

deb (Linux)

### Describe the issue

With webgtk / web2gtk 2.44.2 isntalled the UI shows up as blank and is not able to be interacted with. Unfortunately this includes bringing up the devtools!

I tried removing the following with the same effect

```
~/.config/com.gitbutler.app
~/.local/share/git-butler
~/.local/share/com.gitbutler.app
~/.cache/com.gitbutler.app
~/.cache/git-butler
```

### How to reproduce

Install arch 
Upgrade the following packages
 extra/webkit2gtk      2.44.2-1 -> 2.44.3-1
 extra/webkit2gtk-4.1  2.44.2-1 -> 2.44.3-1
 extra/webkitgtk-6.0   2.44.2-1 -> 2.44.3-1

### Expected behavior

Git butler should work just fine

### Relevant log output

```shell
** (git-butler:10353): WARNING **: 14:36:21.466: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
2024-08-14T05:06:21.525603Z  INFO crates/gitbutler-tauri/src/main.rs:96: starting app version=0.12.19 name=GitButler
2024-08-14T05:06:21.795929Z  INFO list_projects: crates/gitbutler-tauri/src/projects.rs:41: close time.busy=50.4µs time.idle=18.4µs
2024-08-14T05:06:21.800403Z  INFO get_user: crates/gitbutler-tauri/src/users.rs:10: close time.busy=21.7µs time.idle=12.4µs
2024-08-14T05:06:21.884010Z  INFO get_project: crates/gitbutler-tauri/src/projects.rs:32: close time.busy=87.7µs time.idle=15.3µs id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.248330Z  WARN fetch_from_remotes: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a action=Some("auto")
2024-08-14T05:06:22.248345Z  WARN list_branches: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a filter=Some(BranchListingFilter { local: None, applied: None })
2024-08-14T05:06:22.248415Z  WARN git_head: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.248444Z  WARN get_base_branch_data:get_base_branch_data: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.248472Z  INFO git_head: crates/gitbutler-tauri/src/commands.rs:61: close time.busy=3.93ms time.idle=24.0µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.249670Z  INFO get_base_branch_data:get_base_branch_data: crates/gitbutler-branch-actions/src/actions.rs:134: close time.busy=4.79ms time.idle=4.95µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.249692Z  INFO get_base_branch_data: crates/gitbutler-tauri/src/virtual_branches.rs:127: close time.busy=4.85ms time.idle=7.96µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.252473Z  INFO list_branches: crates/gitbutler-tauri/src/virtual_branches.rs:428: close time.busy=9.60ms time.idle=7.97µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a filter=Some(BranchListingFilter { local: None, applied: None })
2024-08-14T05:06:22.299737Z  WARN list_virtual_branches: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.325352Z  INFO list_virtual_branches:workdir: crates/gitbutler-diff/src/diff.rs:128: close time.busy=8.39ms time.idle=16.6µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a commit_oid=5c5c9bc8c920216c7acd004b0cc60917ace27605
2024-08-14T05:06:22.343443Z  INFO list_virtual_branches: crates/gitbutler-tauri/src/virtual_branches.rs:52: close time.busy=44.0ms time.idle=17.1µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a
2024-08-14T05:06:22.412959Z  WARN get_remote_branch_data: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a refname=Remote(Refname { remote: "origin", branch: "ignore-makefiles" })
2024-08-14T05:06:22.414144Z  INFO get_remote_branch_data: crates/gitbutler-tauri/src/virtual_branches.rs:452: close time.busy=1.57ms time.idle=17.0µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a refname=Remote(Refname { remote: "origin", branch: "ignore-makefiles" })
2024-08-14T05:06:22.437675Z  WARN list_branches: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a filter=Some(BranchListingFilter { local: None, applied: None })
2024-08-14T05:06:22.442568Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=82.2µs time.idle=11.6µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.442575Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=2.18µs time.idle=8.31µs name="ignore makefiles"
2024-08-14T05:06:22.443318Z  INFO list_branches: crates/gitbutler-tauri/src/virtual_branches.rs:428: close time.busy=5.96ms time.idle=14.4µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a filter=Some(BranchListingFilter { local: None, applied: None })
2024-08-14T05:06:22.453211Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=4.10µs time.idle=17.9µs name="tests in makefile"
2024-08-14T05:06:22.453309Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=101µs time.idle=19.3µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.456180Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=2.81µs time.idle=4.73µs name="rust/experimentation"
2024-08-14T05:06:22.456209Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=57.4µs time.idle=6.50µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.459162Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=3.14µs time.idle=6.85µs name="origin/ignore-makefiles"
2024-08-14T05:06:22.460253Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=49.1µs time.idle=2.65µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.460768Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=1.76µs time.idle=7.51µs name="tests in makefile"
2024-08-14T05:06:22.460991Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=37.3µs time.idle=2.03µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.461124Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=580ns time.idle=2.29µs name="rust/experimentation"
2024-08-14T05:06:22.461520Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=2.95µs time.idle=5.16µs name="ignore makefiles"
2024-08-14T05:06:22.461527Z  INFO git_get_global_config: crates/gitbutler-tauri/src/commands.rs:93: close time.busy=90.5µs time.idle=4.66µs key="gitbutler.aiModelProvider"
2024-08-14T05:06:22.479718Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=2.89µs time.idle=12.7µs name="tests in makefile"
2024-08-14T05:06:22.486919Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=6.46µs time.idle=21.6µs name="rust/experimentation"
2024-08-14T05:06:22.505575Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=6.51µs time.idle=21.9µs name="origin/ignore-makefiles"
2024-08-14T05:06:22.506581Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=1.65µs time.idle=6.13µs name="tests in makefile"
2024-08-14T05:06:22.506778Z  INFO normalize_branch_name: crates/gitbutler-tauri/src/virtual_branches.rs:23: close time.busy=641ns time.idle=2.40µs name="rust/experimentation"
2024-08-14T05:06:22.886391Z  INFO handle: crates/gitbutler-watcher/src/handler.rs:55: close time.busy=59.1µs time.idle=8.95µs event=GitFileChange(dd7fb8bd-a280-41b7-a572-a632d9715f0a, FETCH_HEAD)
2024-08-14T05:06:22.886198Z  INFO set_project_active:set_project_to_window:file monitor:handle debounced events: crates/gitbutler-watcher/src/file_monitor.rs:119: close time.busy=44.9µs time.idle=26.4µs id=dd7fb8bd-a280-41b7-a572-a632d9715f0a window="main" project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a fs_events=99 fs_events=99 ignored=0 ignored=0 git_noop=2 git_noop=2 git=1 git=1 project=0 project=0 git_dedup=1 git_dedup=1
2024-08-14T05:06:25.228647Z  WARN fetch_from_remotes:get_base_branch_data: crates/gitbutler-command-context/src/lib.rs:31: failed to get gitbutler.didSetPrune for repository at <redacted>; cannot disable gc: config value 'gitbutler.didSetPrune' was not found; class=Config (7); code=NotFound (-3) project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a action=Some("auto")
2024-08-14T05:06:25.229344Z  INFO fetch_from_remotes:get_base_branch_data: crates/gitbutler-branch-actions/src/actions.rs:134: close time.busy=828µs time.idle=12.7µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a action=Some("auto")
2024-08-14T05:06:25.229359Z  INFO fetch_from_remotes: crates/gitbutler-tauri/src/virtual_branches.rs:480: close time.busy=2.98s time.idle=6.03µs project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a action=Some("auto")
2024-08-14T05:06:25.387152Z  INFO set_project_active:set_project_to_window:file monitor:handle debounced events: crates/gitbutler-watcher/src/file_monitor.rs:119: close time.busy=39.4µs time.idle=28.0µs id=dd7fb8bd-a280-41b7-a572-a632d9715f0a window="main" project_id=dd7fb8bd-a280-41b7-a572-a632d9715f0a fs_events=8 fs_events=8 ignored=0 ignored=0 git_noop=0 git_noop=0 git=1 git=1 project=0 project=0 git_dedup=1 git_dedup=1
2024-08-14T05:06:25.387392Z  INFO handle: crates/gitbutler-watcher/src/handler.rs:55: close time.busy=113µs time.idle=20.5µs event=GitFileChange(dd7fb8bd-a280-41b7-a572-a632d9715f0a, FETCH_HEAD)
```

**Comment by ndom91** [comment]
Hey @Psykar thanks for pointing this out, I'll take a closer look.

When you say blank you also mean the entire file menu? (File, Edit, Help, etc.)?

**Comment by ygorluiz** [comment]
I believe this is what he means
![screenshot_14082024_114250](https://github.com/user-attachments/assets/4b9dfc1d-6eb4-403a-b745-d876d80bfaaa)

**Comment by ygorluiz** [comment]
@ndom91 We cant open dev tools,This happens after clicking continue on the home screen. 
![screenshot_14082024_121603](https://github.com/user-attachments/assets/b8936f6f-2c67-415e-8e4b-70bc0d36ed81)

**Comment by ndom91** [comment]
@ygorluiz maybe I'm being unintentionally thick, but what do you mean by "we can't open dev tools". Can you be more specific?

Is the option not visible for you? The dev tools should be available under "View" -> "Developer Tools"

**Comment by ygorluiz** [comment]
@ndom91 it is available and clickable but it does not open.

**Comment by 7RST1** [comment]
I have confirmed this issue on my Manjaro Linux

**Comment by Psykar** [comment]
Looks like this was an upstream issue not a gitbutler issue.
https://gitlab.archlinux.org/archlinux/packaging/packages/webkit2gtk-4.1/-/issues/1

**Comment by ndom91** [comment]
Thanks for the info folks. The arch gitlab issue mentions their specific issue should be fixed in `2.44.3-2`. Can you give that a shot and let us know if gitbutler also works as expected with that version? 🙏

**Comment by Psykar** [comment]
Yeah it works for me now hence why I closed it :)
